### DISCUSSION TITLE
为什么 IPv6 难以取代 IPv4 - 面向信仰编程 · /whys-the-design-ipv6-replacing-ipv4

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#613] [Announcements]
https://draveness.me/whys-the-design-ipv6-replacing-ipv4/

**Comment by thoamsy** [comment]
有个疑问，IPv6 除了用不完相比 IPv4 的优点还有哪些呢？另外如果 6 完全普及后，是否对非企业网络 NAT 都可以不用了

**Comment by draveness** [comment]
> 有个疑问，IPv6 除了用不完相比 IPv4 的优点还有哪些呢？

+ IPv6 的性能比 IPv4 更好一些；
+ IPv6 可以在网络层提供加密，不过好像不是必选的；
+ ...

这个资料还是挺多的，可以搜一下

> 另外如果 6 完全普及后，是否对非企业网络 NAT 都可以不用了

可以肯定是可以，但是 NAT 可以避免内部 IP 的频繁修改，也可以当做防火墙保护内部网络

**Comment by rocxlp** [comment]
国内前100APP今年要支持IPv6，要考核的，并不是说说看的。
所以国内推进力度还是蛮强的

**Comment by draveness** [comment]
> 国内前100APP今年要支持IPv6，要考核的，并不是说说看的。
> 所以国内推进力度还是蛮强的

确实有相关的政策，不过还是看看落地情况。

### 相关信息

+ [工业和信息化部关于开展2020年IPv6端到端贯通能力提升专项行动的通知](http://www.miit.gov.cn/n1146285/n1146352/n3054355/n3057674/n4704636/c7830783/content.html)

**Comment by zhubo006** [comment]
查了一下还真有IPv5  🙃

> 我们说的IPv4和IPv6中的“4”和“6”都是指的IP协议的版本号，IPv4全称叫Internet Protocol Version 4,意思是IP协议第四版，属于网络TCP/IP中的网络层（IP层）的网络协议版本。历史上的确有IPv5协议，但是从TCP/IP网络层的协议结构层次而言，IPv5协议依旧基于IPv4的协议框架，不能算是和IPv4、IPv6并列基础IP协议架构体系，所以我们可以认为，从IP协议的版本来看，IPv5不存在
> https://www.sohu.com/a/326904209_120197186

**Comment by draveness** [comment]
> 查了一下还真有IPv5  🙃
> 
> > 我们说的IPv4和IPv6中的“4”和“6”都是指的IP协议的版本号，IPv4全称叫Internet Protocol Version 4,意思是IP协议第四版，属于网络TCP/IP中的网络层（IP层）的网络协议版本。历史上的确有IPv5协议，但是从TCP/IP网络层的协议结构层次而言，IPv5协议依旧基于IPv4的协议框架，不能算是和IPv4、IPv6并列基础IP协议架构体系，所以我们可以认为，从IP协议的版本来看，IPv5不存在
> > https://www.sohu.com/a/326904209_120197186

可以再看看 IPv 0123🙃

**Comment by zhubo006** [comment]
> @draveness
> > 查了一下还真有IPv5  🙃
> > 
> > > 我们说的IPv4和IPv6中的“4”和“6”都是指的IP协议的版本号，IPv4全称叫Internet Protocol Version 4,意思是IP协议第四版，属于网络TCP/IP中的网络层（IP层）的网络协议版本。历史上的确有IPv5协议，但是从TCP/IP网络层的协议结构层次而言，IPv5协议依旧基于IPv4的协议框架，不能算是和IPv4、IPv6并列基础IP协议架构体系，所以我们可以认为，从IP协议的版本来看，IPv5不存在
> > > https://www.sohu.com/a/326904209_120197186
> 
> 可以再看看 IPv 0123🙃

> Where is IPv1, 2, 3,and 5?
> https://blog.alertlogic.com/blog/where-is-ipv1,-2,-3,and-5/

**Comment by Liutos** [comment]
想请教一下博主，文章最后的微信名片是怎么制作出来的？

**Comment by draveness** [comment]
> 想请教一下博主，文章最后的微信名片是怎么制作出来的？

你注册公众号就有了

**Comment by Liutos** [comment]
> @draveness
> > 想请教一下博主，文章最后的微信名片是怎么制作出来的？
> 
> 你注册公众号就有了

好的我去找一下😂

**Comment by rocxlp** [comment]
> @draveness
> > 国内前100APP今年要支持IPv6，要考核的，并不是说说看的。
> > 所以国内推进力度还是蛮强的
> 
> 确实有相关的政策，不过还是看看落地情况。
> 
> ### 相关信息
> 
> + [工业和信息化部关于开展2020年IPv6端到端贯通能力提升专项行动的通知](http://www.miit.gov.cn/n1146285/n1146352/n3054355/n3057674/n4704636/c7830783/content.html)

落地没啥问题，我所在100名靠后的，今年再搞这个东西，快要全网上线；我家动作还比较慢，不少已经落实好了。

**Comment by vincentor** [comment]
> 虽然 IP 协议的设计者承认 IPv6 没有实现向前兼容是最大的错误

这里应该是指向后兼容？

**Comment by draveness** [comment]
> > 虽然 IP 协议的设计者承认 IPv6 没有实现向前兼容是最大的错误
> 
> 这里应该是指向后兼容？

~~是 Forward~~

---

*2020-06-08 UPDATES: 向前 -> 向后兼容*

**Comment by vincentor** [comment]
> @draveness
> > > 虽然 IP 协议的设计者承认 IPv6 没有实现向前兼容是最大的错误
> > 
> > 这里应该是指向后兼容？
> 
> 是 Forward

可是你引述的文章标题是 Biggest mistake for IPv6: It’s not backwards compatible

**Comment by draveness** [comment]
> > @draveness
> > > > 虽然 IP 协议的设计者承认 IPv6 没有实现向前兼容是最大的错误
> > > 
> > > 
> > > 这里应该是指向后兼容？
> > 
> > 
> > 是 Forward
> 
> 可是你引述的文章标题是 Biggest mistake for IPv6: It’s not backwards compatible

昂，确实是这里的问题

**Comment by jacoesad** [comment]
前几天在微信看到这篇文章，原来原创在这了！

**Comment by jolin20082** [comment]
这个网站是用什么架设的？版面很不错，简洁大方
